Applications/IdentityDeletionJobs/src/Job.IdentityDeletion/Workers/ActualDeletionWorker.cs

```diff
@@ -70,6 +76,10 @@ private async Task ExecuteDeletion(IdentityAddress identityAddress, Cancellation
 
     private async Task NotifyIdentityAboutStartingDeletion(IdentityAddress identityAddress, CancellationToken cancellationToken)
     {
+        var identity = await _identitiesRepository.FindByAddress(identityAddress, CancellationToken.None) ?? throw new NotFoundException(nameof(Identity));
+
+        identity.EnsureIdentityIsToBeDeleted();
```

Comment:
Why did you add this?

Reply:
Well I thought this also should be blocked... Changes reverted.